test/common/router/router_test.cc

```diff
@@ -275,6 +278,33 @@ class RouterTestSuppressEnvoyHeaders : public RouterTestBase {
       : RouterTestBase(false, true, Protobuf::RepeatedPtrField<std::string>{}) {}
 };
 
+TEST_F(RouterTest, UpdateFilterState) {
```

Comment:
This feature needs an integration test. Please add a TLS integration test which exercises this.

Reply:
@mattklein123 Is there a helpful test suite related with this?

Reply:
See `TEST_P(ProxyFilterIntegrationTest, UpstreamTls)` for a test which verifies that SNI is set properly. You will need to have a very similar test.